query/src/servers/http/v1/load.rs

```diff
@@ -0,0 +1,108 @@
+// Copyright 2021 Datafuse Labs.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use std::sync::Arc;
+
+use async_compat::CompatExt;
+use async_stream::stream;
+use common_planners::PlanNode;
+use common_streams::CsvSource;
+use common_streams::Source;
+use futures::StreamExt;
+use poem::error::BadRequest;
+use poem::error::Result as PoemResult;
+use poem::web::Data;
+use poem::web::Json;
+use poem::web::Multipart;
+use poem::Request;
+use serde::Deserialize;
+use serde::Serialize;
+
+use crate::interpreters::InterpreterFactory;
+use crate::sessions::SessionManager;
+use crate::sql::PlanParser;
+
+#[derive(Serialize, Deserialize, Debug)]
+pub struct LoadResponse {
+    pub id: String,
+    pub state: String,
+    pub error: Option<String>,
+}
+
+#[poem::handler]
+pub async fn streaming_load(
+    req: &Request,
+    mut multipart: Multipart,
+    sessions_extension: Data<&Arc<SessionManager>>,
+) -> PoemResult<Json<LoadResponse>> {
+    let session_manager = sessions_extension.0;
+    let session = session_manager.create_session("Streaming load")?;
+    let context = session.create_context().await?;
+    let insert_sql = req
+        .headers()
+        .get("insert_sql")
+        .and_then(|v| v.to_str().ok())
+        .unwrap_or("");
+
+    let csv_header = req
+        .headers()
+        .get("csv_header")
+        .and_then(|v| v.to_str().ok())
+        .unwrap_or("0")
+        .eq_ignore_ascii_case("1");
+
+    let plan = PlanParser::parse(insert_sql, context.clone()).await?;
+    let max_block_size = context.get_settings().get_max_block_size()? as usize;
+    let interpreter = InterpreterFactory::get(context.clone(), plan.clone())?;
+
+    if let PlanNode::InsertInto(insert) = &plan {
+        if insert.format.is_none() {
+            return Err(BadRequest(
+                "Streaming load only support csv format, but got none",
+            ));
+        }
+
+        if insert.format.as_ref().unwrap().to_lowercase().as_str() != "csv" {
+            return Err(BadRequest("Streaming load only support csv format"));
+        }
+    }
+
+    let stream = stream! {
+        while let Ok(Some(field)) = multipart.next_field().await {
```

Comment:
why need multipart?

Reply:
multipart supports multi files to upload.

Reply:
got it, thank you